packages/integration_test/lib/integration_test_driver_extended.dart

```diff
@@ -28,11 +31,12 @@ Future<void> integrationDriver(
       response.data!['web_driver_command'] != '${WebDriverCommandType.noop}') {
     final String? webDriverCommand = response.data!['web_driver_command'] as String?;
     if (webDriverCommand == '${WebDriverCommandType.screenshot}') {
+      assert(onScreenshot != null, 'screenshot command requires an onScreenshot callback');
       // Use `driver.screenshot()` method to get a screenshot of the web page.
       final List<int> screenshotImage = await driver.screenshot();
-      final String? screenshotName = response.data!['screenshot_name'] as String?;
+      final String screenshotName = (response.data!['screenshot_name'] ?? '') as String;
```

Comment:
question: is this the correct approach (will it ever be '')?

Reply:
Yeah, probably not. I was taking my cue from the fact that it was set up as nullable, but the `onScreenshot` signature doesn't accept a null name anyhow.

Thanks for pointing that out.